rotkehlchen/api/rest.py

```diff
@@ -280,6 +285,49 @@ def inner(rest_api: 'RestAPI', async_query: bool = False, **kwargs: Any) -> Resp
     return wrapper
 
 
+def check_cached_image(image_path: Path, match_header: Optional[str]) -> Optional[Response]:
+    """
+    Check whether md5 hash of the file at `image_path` matches the value in `match_header`.
+    If it does -- returns response with status code NOT_MODIFIED.
+    """
+    md5_hash = file_md5(image_path)
+    if md5_hash and match_header and match_header == md5_hash:
+        # Response content unmodified
+        return make_response(
+            (
+                b'',
+                HTTPStatus.NOT_MODIFIED,
+                {'mimetype': 'image/png', 'Content-Type': 'image/png'},
+            ),
+        )
+
+    return None
+
+
+def maybe_return_image(image_path: Optional[Path]) -> Response:
+    """
+    Checks whether file at `image_path` exists and if it does -- returns this file.
```

Comment:
```suggestion
    """Checks whether the file at `image_path` exists.
    
    Returns a response with the image if it exists, otherwise a NOT FOUND response.
    """
```